test/test_navierstokes.py

```diff
@@ -0,0 +1,1040 @@
+                              momentum=momentum), p, temperature
+
+
+class TrigSolution1(FluidManufacturedSolution):
+    """CNS manufactured solution designed to vanish on the domain boundary."""
+
+    def __init__(self, dim, q_coeff, x_coeff, lx=None,
+                 gamma=1.4, gas_const=287.):
+        """Initialize it."""
+        super().__init__(dim, lx, gamma, gas_const)
+        self._q_coeff = q_coeff
+
+    def get_solution(self, x, t):
+        """Return the symbolically-compatible solution."""
+        velocity = 0
+        press = 1
+        density = 1
+
+        mom = density*velocity
+        temperature = press/(density*self._gas_const)
+        energy = press/(self._gamma - 1) + density*np.dot(velocity, velocity)
+        return make_conserved(dim=self._dim, mass=density, momentum=mom,
+                              energy=energy), press, temperature
+
+    def get_mesh(self, x, t):
+        """Get the mesh."""
+        return super().get_mesh(x, t)
+
+    def get_boundaries(self):
+        """Get the boundaries."""
+        return super().get_boundaries()
+
+
+class TestSolution(FluidManufacturedSolution):
+    """Trivial manufactured solution."""
+
+    def __init__(self, dim=2, density=1, pressure=1, velocity=None):
+        """Init the man soln."""
+        super().__init__(dim)
+        if velocity is None:
+            velocity = make_obj_array([0 for _ in range(dim)])
+        assert len(velocity) == dim
+        self._vel = velocity
+        self._rho = density
+        self._pressure = pressure
+
+    def get_mesh(self, n):
+        """Get the mesh."""
+        return super().get_mesh(n)
+
+    def get_boundaries(self, discr, actx, t):
+        """Get the boundaries."""
+        return super().get_boundaries(discr, actx, t)
+
+    def get_solution(self, x, t):
+        """Return sym soln."""
+        density = 1*x[0]
+        energy = 2*x[1]**2
+        mom = make_obj_array([i*x[0]*x[1] for i in range(self._dim)])
+        pressure = x[0]*x[0]*x[0]
+        temperature = x[1]*x[1]*x[1]
+
+        return make_conserved(dim=self._dim, mass=density, momentum=mom,
+                              energy=energy), pressure, temperature
+
+
+# @pytest.mark.parametrize("nspecies", [0, 10])
+@pytest.mark.parametrize("order", [1, 2, 3])
+@pytest.mark.parametrize(("dim", "manufactured_soln", "mu"),
+                         [(1, UniformSolution(dim=1), 0),
+                          (2, UniformSolution(dim=2), 0),
+                          (3, UniformSolution(dim=3), 0),
+                          (2, UniformSolution(dim=2, velocity=[1, 1]), 0),
+                          (3, UniformSolution(dim=3, velocity=[1, 1, 1]), 0),
+                          (2, IsentropicVortex(), 0),
+                          (2, IsentropicVortex(velocity=[1, 1]), 0),
+                          (2, ShearFlow(mu=.01, gamma=3/2), .01)])
+def test_exact_mms(actx_factory, order, dim, manufactured_soln, mu):
```

Comment:
Does this belong in `test_operators.py`?

Reply:
Er... maybe?   `test_operators.py` was supposed to test the operator wrappers stuff in `operators.py`, but we can move it if you think it should move.

Reply:
Just a question... (shouldn't hold up the merge)

Reply:
Sure.  I am happy to go with consensus on this.   I don't have any strong feelings about it.  We could make a dedicated testing module to test the symbolic fluid operators themselves, too.   Maybe that would be more appropriate?

Reply:
Yeah I'm not sure if `test_operators.py` would be the right place for it. I would be OK with leaving it where it is, but a new module (or even `test_symbolic.py`) would also be fine.

Reply:
I guess if it's only used here, then I'd say leave it.

Reply:
Just fyi, i did toy around with moving it off to its own  place (`test_symbolic_fluid.py`) but it was a little awkward to do that because they share infrastructure, and there was really no other natural place to drop that infrastructure.  We can work around it if that is desired - but I am also OK with leaving it as-is.   

@inducer and I have discussed that we would like to meld the symbolic manufactured solution constructs with the actual fluid initializers - and that would be a natural time to address this particular concern - but that is going to need to be tackled in its own PR.  It was not straightforward to do when I tried it previously.